src/base/ops.rs

```diff
@@ -398,11 +398,13 @@ where
     /// iter::empty::<DMatrix<f64>>().sum::<DMatrix<f64>>(); // panics!
     /// ```
     fn sum<I: Iterator<Item = OMatrix<T, Dyn, C>>>(mut iter: I) -> OMatrix<T, Dyn, C> {
-        if let Some(first) = iter.next() {
-            iter.fold(first, |acc, x| acc + x)
-        } else {
-            panic!("Cannot compute `sum` of empty iterator.")
+        let mut ret = iter
+            .next()
+            .expect("Cannot compute `sum` of empty iterator.");
+        while let Some(next) = iter.next() {
+            ret += next;
         }
+        ret
```

Comment:
Why not retain the `fold`?

Reply:
so that we use add_assign as opposed to allocate a new matrix every time.